lib/active_admin/filters/active_filter.rb

```diff
@@ -75,13 +89,34 @@ def find_class?
       def find_class
         if condition_attribute.klass != resource_class && condition_attribute.klass.primary_key == name.to_s
           condition_attribute.klass
-        else
-          assoc = condition_attribute.klass.reflect_on_all_associations.
-            reject { |r| r.options[:polymorphic] }. #skip polymorphic
-            detect { |r| r.foreign_key.to_s == name.to_s }
-          assoc.class_name.constantize if assoc
+        elsif predicate_association
+          predicate_association.class_name.constantize
+        end
+      end
+
+      def filter
+        resource.filters[name.to_sym]
+      end
+
+      def related_primary_key
+        if predicate_association
+          predicate_association.active_record_primary_key
+        elsif related_class
+          related_class.primary_key
```

Comment:
Personally I like terse boolean operators over if/else:
```ruby
      def related_primary_key
        predicate_association && predicate_association.active_record_primary_key ||
          related_class && related_class.primary_key
      end
```

Reply:
Hurts readability though. You'll still need a block to return the proper primary key (as that only returns a `true` or `false`.

Reply:
It's a personal preference, maybe from writing too much C or perl. :-)

Reply:
Definitely understand. I think I leave it as is for now though. It'll help me understand what's going on more quickly if I look at in a month than chained binary operators.  :see_no_evil: